infrastructure/transform/main.tf

```diff
@@ -109,19 +105,41 @@ resource "azurerm_role_assignment" "adf_can_access_kv_secrets" {
   principal_id         = azurerm_data_factory.adf.identity[0].principal_id
 }
 
+# Get all directories that have activities.json in them and say that they are pipeline directories
+resource "azurerm_data_factory_pipeline" "pipeline" {
+  for_each        = local.pipeline_dirs
+  name            = "databricks-pipeline-${basename(each.value)}-${var.naming_suffix}"
+  data_factory_id = azurerm_data_factory.adf.id
+  activities_json = file("${each.value}/${local.activities_file}")
+
+  depends_on = [
+    azurerm_data_factory_linked_service_azure_databricks.msi_linked
+  ]
+}
+
+# Assuming that all artifacts will be built
+resource "databricks_dbfs_file" "dbfs_artifact_upload" {
+  for_each = { for artifact in local.artifacts : artifact.artifact_path => artifact.pipeline }
+  # Source path on local filesystem
+  source = each.key
+  # Path on DBFS
+  path = "/pipelines/${each.value}/${local.artifacts_dir}/${basename(each.key)}"
+}
+
 resource "azurerm_data_factory_linked_service_azure_databricks" "msi_linked" {
   name            = "ADBLinkedServiceViaMSI"
   data_factory_id = azurerm_data_factory.adf.id
   description     = "Azure Databricks linked service via MSI"
   adb_domain      = "https://${azurerm_databricks_workspace.databricks.workspace_url}"
 
   msi_work_space_resource_id = azurerm_databricks_workspace.databricks.id
-  existing_cluster_id        = databricks_cluster.fixed_single_node.cluster_id
+
+  existing_cluster_id = databricks_cluster.fixed_single_node.cluster_id
 }
 
 resource "azurerm_data_factory_linked_service_key_vault" "msi_linked" {
   name            = "KVLinkedServiceViaMSI"
   data_factory_id = azurerm_data_factory.adf.id
   description     = "Key Vault linked service via MSI"
   key_vault_id    = var.core_kv_id
-}
+}
```

Comment:
nit: a few missing end of file newlines

Reply:
wonder why these aren't caught by a linter...

Reply:
an excellent point